### ISSUE TITLE
Define "primary key" for objects

### ISSUE THREAD

**Issue by ralfhandl** [OAI/OpenAPI-Specification#587]
OData APIs are based on an entity-relationship model, so the top-level resources are collections of identically structured objects, and there's a subset of properties whose values uniquely identify an object (think: records in a database table).

A similar issue has been addressed in the JSON Schema discussion group, see https://groups.google.com/d/msg/json-schema/CYd8xk1Czyg/XLNxEzDOPj4J 

We currently solve this with an extension keyword "x-keys" whose value is an array of strings listing the names of the "primary key" properties.

Anyone else seeing a need for this and wanting to have it in OpenAPI.Next?

**Comment by webron** [comment]
Tackling PR: #741

**Comment by avkonst** [comment]
Yes, I do the same with x-keys.

**Comment by ePaul** [comment]
#741 (or #894) did not contain anything for this issue.

**Comment by webron** [comment]
Yup, not everything was resolved in #741, and this is why this ticket was not closed.

**Comment by demonking** [comment]
@ralfhandl something new there? or should we use an extension ?
